File: replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function(arr) {
    
    let max = -1
    
    for (let i = arr.length-1; i >= 0; i--){
        current = arr[i]
        arr[i] = max
        max = Math.max(current, max)
    }
    return arr
    
};

File: replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.rs
impl Solution {
    pub fn replace_elements(mut arr: Vec<i32>) -> Vec<i32> {
        
        let mut max = -1;
        
        for num in arr.iter_mut().rev(){
            let current = *num;
            *num = max;
            max = current.max(max);
        }
        arr
    }
}